### PR TITLE
improve debug detection features

### PR DESCRIPTION
**Pull request by mr-tz**
ref #595

### CODE REVIEW

File: anti-analysis/anti-debugging/debugger-detection/check-for-debugger-via-api.yml [hunk 1]
```diff
@@ -15,6 +15,9 @@ rule:
       - al-khaser_x86.exe_:0x420000
   features:
     - or:
+      # We're not including kernel32.IsDebuggerPresent here because some exception handlers and other compiler-inserted
+      # code may add calls to it, especially in debug builds. So, likely even with pretty good library code detection
+      # this feature could result in many false positives.
```
Comment (mr-tz) [inline anti-analysis/anti-debugging/debugger-detection/check-for-debugger-via-api.yml:20]: see https://github.com/mandiant/capa-rules/pull/100#discussion_r470065231

File: host-interaction/process/list/enumerate-processes-via-ntquerysysteminformation.yml [hunk 2]
```diff
@@ -13,4 +13,8 @@ rule:
   features:
     - and:
       - number: 0x5 = SYSTEM_PROCESS_INFORMATION
-      - api: NtQuerySystemInformation
+      - or:
+        - api: NtQuerySystemInformation
+        - api: NtQuerySystemInformationEx
+        - api: ZwQuerySystemInformation
+        - api: ZwQuerySystemInformationEx
```
Comment (mr-tz) [inline host-interaction/process/list/enumerate-processes-via-ntquerysysteminformation.yml:20]: see #720

File: nursery/check-systemkerneldebuggerinformation.yml [hunk 3]
```diff
@@ -6,10 +6,14 @@ rule:
...
     mbc:
-      - Anti-Behavioral Analysis::Debugger Detection::NtQueryInformationProcess [B0001.012]
+      - Anti-Behavioral Analysis::Debugger Detection [B0001]
     references:
       - https://github.com/LordNoteworthy/al-khaser/blob/master/al-khaser/AntiDebug/NtQuerySystemInformation_SystemKernelDebuggerInformation.cpp
   features:
     - and:
-      - api: NtQueryInformationProcess
+      - or:
+        - api: NtQuerySystemInformation
```
Comment (mr-tz) [inline nursery/check-systemkerneldebuggerinformation.yml:15]: fix: `...SystemInformation` (see al-khaser and Anti-DebugNet source)

File: anti-analysis/anti-debugging/debugger-detection/check-for-debugger-via-api.yml (hunk 1 above)
Comment (mike-hunhoff) [reply anti-analysis/anti-debugging/debugger-detection/check-for-debugger-via-api.yml:20]: nice!

**Pull request merged**